package/MDAnalysis/coordinates/RDKit.py

```diff
@@ -339,108 +322,124 @@ def convert(self, obj, cache=True, NoImplicit=True, max_iter=200):
-                "orders to structures with implicit hydrogens. Alternatively, "
-                "you can use the parameter `NoImplicit=False` when using the "
-                "converter to allow implicit hydrogens and disable inferring "
-                "bond orders and charges."
-            )
-
-        # attributes accepted in PDBResidueInfo object
-        pdb_attrs = {}
-        if hasattr(ag, "bfactors") and hasattr(ag, "tempfactors"):
-            raise AttributeError(
-                "Both `tempfactors` and `bfactors` attributes are present but "
-                "only one can be assigned to the RDKit molecule. Please "
-                "delete the unnecessary one and retry."
-            )
-        for attr in RDATTRIBUTES.keys():
-            if hasattr(ag, attr):
-                pdb_attrs[attr] = getattr(ag, attr)
-
-        other_attrs = {}
-        for attr in ["charges", "segids", "types"]:
-            if hasattr(ag, attr):
-                other_attrs[attr] = getattr(ag, attr)
-
-        mol = Chem.RWMol()
-        # map index in universe to index in mol
-        atom_mapper = {}
-
-        for i, (atom, element) in enumerate(zip(ag, elements)):
-            # create atom
-            rdatom = Chem.Atom(element.capitalize())
-            # enable/disable adding implicit H to the molecule
-            rdatom.SetNoImplicit(NoImplicit)
-            # add PDB-like properties
-            mi = Chem.AtomPDBResidueInfo()
-            for attr, values in pdb_attrs.items():
-                _add_mda_attr_to_rdkit(attr, values[i], mi)
-            rdatom.SetMonomerInfo(mi)
-            # other properties
-            for attr in other_attrs.keys():
-                value = other_attrs[attr][i]
-                attr = "_MDAnalysis_%s" % _TOPOLOGY_ATTRS[attr].singular
-                _set_atom_property(rdatom, attr, value)
-            _set_atom_property(rdatom, "_MDAnalysis_index", i)
-            # add atom
-            index = mol.AddAtom(rdatom)
-            atom_mapper[atom.ix] = index
+@lru_cache(maxsize=2)
+def atomgroup_to_mol(ag, NoImplicit=True, max_iter=200):
+    """Converts an AtomGroup to an RDKit molecule without coordinates.
 
+    Parameters
+    -----------
+    ag : MDAnalysis.core.groups.AtomGroup
+        The AtomGroup to convert
+    NoImplicit : bool
+        Prevent adding hydrogens to the molecule
+    max_iter : int
+        Maximum number of iterations to standardize conjugated systems.
+        See :func:`_rebuild_conjugated_bonds`
+    """
+    try:
+        elements = ag.elements
+    except NoDataError:
+        raise AttributeError(
+            "The `elements` attribute is required for the RDKitConverter "
+            "but is not present in this AtomGroup. Please refer to the "
+            "documentation to guess elements from other attributes or "
+            "type `help(mda.topology.guessers)`") from None
+
+    if "H" not in ag.elements:
```

Comment:
Maybe this should raise an exception and have an explicit opt-in kwarg (like "force")? Can you think of a case where one would want this? ... How often do we have structures without hydrogens?

Reply:
@orbeckst PDB files often don't have them all explicit. There are also the rather rare cases of small molecules with no hydrogen like CO2 and so on...
I'm not too sure if it's desirable to return a molecule and just raise a warning even if the conversion will output something that is most likely 💩

Reply:
> I'm not too sure if it's desirable to return a molecule and just raise a warning even if the conversion will output something that is most likely 💩

No, I expect that nobody reads warnings.

If something will likely be wrong then better fail.

That should be the guiding principle. If there are special cases then maybe provide a kwarg for power users to override the safety check. It will be a bit annoying when you work with inorganic molecules (or CO2) but presumably these will be a minority and the exception message could state clearly what to do if the user thinks that this should work.